code/datums/components/material_container.dm

```diff
@@ -31,12 +43,44 @@
 			var/mat_path = possible_mats[id]
 			materials[id] = new mat_path()
 
-/datum/material_container/Destroy()
-	owner = null
-	return ..()
+/datum/component/material_container/proc/OnExamine(mob/user)
+	for(var/I in materials)
+		var/datum/material/M = materials[I]
+		to_chat(user, "<span class='notice'>It has [amount(M.id)] units of [lowertext(M.name)] stored.</span>")
+
+/datum/component/material_container/proc/OnAttackBy(obj/item/I, mob/living/user)
```

Comment:
Majority of duplicated attackby code now goes here, reducing copypasta all around